### ISSUE TITLE
Podman Desktop Hangs at "Initializing Podman" on Mac Book M1

### ISSUE THREAD

**Issue by myfear** [containers/podman-desktop#1633]
### Bug description

![podmanstuck](https://user-images.githubusercontent.com/1358554/223731030-0fe397f1-47d7-4bed-b528-a760f00ae53f.jpg)

Podman Desktop stuck in above state after install.

### Operating system

macOS Ventura 13..2.1

### Version

0.12.0

### Steps to reproduce

Install Podman Desktop on a blank system.

you can reproduce the hang on initialize podman by deleting all the podman artifacts[1] and begining again with the installer disk image -> drag to applications -> run -> install podman -> quit -> restart podman desktop -> intialize podman -> HANG

[1]: 
~ % rm -r .config/containers 
~ % rm -rf .local/share/containers       
~ % sudo rm -rf /opt/podman 
~ % rm -rf Library/Application\ Support/Podman\ Desktop 
~ % sudo rm -r /private/etc/paths.d/podman-pkg 
~ % sudo rm -r /usr/local/podman
~ % sudo rm /Library/LaunchDaemons/com.github.containers.podman.helper-josh.plist

~ % mdfind -name podman|grep -v 'devel\/go' /Users/josh/Downloads/podman-desktop-0.12.0-arm64.dmg

**Comment by joshix** [comment]
Workaround is something like: At HANG T+?, quit Podman Desktop. Relaunch Podman Desktop to find ready state.

NB: HANG here and above represents no progress no completion of podman init process, but not a hang or unresponsiveness in the electron UI side.

**Comment by benoitf** [comment]
It could be interesting to know what is happening when you do in cli 'podman machine init' step and then 'podman machine starts step ( instead of launching podman desktop and clicking on the toggle to initiale podman )

Do you have errors ?

**Comment by joshix** [comment]
I used the arm64 macos .pkg from [podman-installer-macos-arm64.pkg](https://github.com/containers/podman/releases/download/v4.4.2/podman-installer-macos-arm64.pkg).

```
j@s2 ~ % podman machine init
Downloading VM image: fedora-coreos-37.20230303.2.0-qemu.aarch64.qcow2.xz: done
Extracting compressed file
Error: key generation failed, unable to read from stderr: exit status 1
j@s2 ~ % podman machine start
Starting machine "podman-machine-default"
WARN[0000] podman helper is installed, but was not able to claim the global docker sock 
Waiting for VM ...
Error: qemu exited unexpectedly with exit code 1, stderr: qemu-system-aarch64: -drive file=/Users/j/.local/share/containers/podman/machine/qemu/podman-machine-default_ovmf_vars.fd,if=pflash,format=raw: Could not open '/Users/j/.local/share/containers/podman/machine/qemu/podman-machine-default_ovmf_vars.fd': No such file or directory
```

Note that _sometimes_ -- cannot reliably repeat -- the GUI displays the same error `key generation failed` when the init step initiated in the GUI is spinning.

Note that I removed `~/.local/share/containers/podman/machine/qemu/podman-machine-default_ovmf_vars.fd` manually when cleaning the environment to reproduce the GUI spinning at init step. It doesn't seem to get recreated on a subsequent install, so evidently there is state still lingering somewhere telling us it is not the first run?

**Comment by TomasHubelbauer** [comment]
I can confirm this happens to me as well. Podman 0.12.0 and macOS Ventura 13.3. The workaround is the same as others already mentioned: wait a few minutes, restart Podman Desktop. I checked `podman machine list` first before restarting Podman Desktop to see the default machine got created so I was reasonably sure that after the restart everything would work fine. BTW there were no errors in the developer tools console before I restarted Podman Desktop.

This was already raised in #951 and supposedly fixed in #953 but I think the fix was only partial seeing as it continues to happen.

**Comment by benoitf** [comment]
if you are stuck please provide logs ( View / toggle developer tools and then console and paste all the logs)
There is an error thrown somewhere

**Comment by TomasHubelbauer** [comment]
As I've already started, there was no error thrown in the developer tools. Unfortunately I have not saved the contents of the console. After restarting the Podman Desktop app, the VM was discovered and the app worked, so I cannot reproduce this issue anymore to get the logs.

**Comment by benoitf** [comment]
this is the main problem to fix without knowing the root cause. If you're able to reproduce please report the logs. Thanks

**Comment by TomasHubelbauer** [comment]
Have you tried installing Podman Desktop on a clean macOS install? I might try uninstalling and reinstalling it later if that's enough to reproduce the issue. BTW I used Homebrew to install both Podman and Podman Desktop if that helps. I will let you know if I've managed to reproduce the error on my same machine after uninstalling and reinstalling Podman.

**Comment by benoitf** [comment]
@TomasHubelbauer let us know. Thanks

**Comment by ST33LDI9ITAL** [comment]
Same situation. Fresh install from website. Asked to install podman, so did, it didn't refresh detection and switch to initialize.. so had to restart Podman Desktop, (filed separate ticket for that). Once quit and went back into PD, hit initialize and start, then got a good ol never ending infinite loading loop.

<img width="1047" alt="image" src="https://user-images.githubusercontent.com/12997114/233738482-593c9174-2d1f-4f78-85c0-5542bbba6019.png">

**Comment by sgallagher** [comment]
Same as @ST33LDI9ITAL  above, on Podman Desktop 0.15.0 and Podman Engine 4.5.0.

**Comment by kwiatks** [comment]
Same - Podman Desktop 1.0.1, Engine 4.5.0
![image](https://github.com/containers/podman-desktop/assets/3601821/e005c06d-ef86-412f-8fdc-154e3121fb12)
and
![image](https://github.com/containers/podman-desktop/assets/3601821/650e16e7-23dc-4237-8ab1-020f5f5e451d)

**Comment by TomasHubelbauer** [comment]
I was able to try this again on a new machine. I am using macOS Ventura 13.4 (latest) on an MX series CPU. Docker was already installed and running on the machine. I installed Podman according to the instructions on https://podman-desktop.io/docs/Installation/macos-install and went through the process of fixing the detection checks (not being in PATH) as well, the other prerequisites were all met and the application is displaying that Docker is running:

<img width="698" alt="image" src="https://github.com/containers/podman-desktop/assets/6831144/75fb64e5-d454-486f-a6e4-09ebd8601a03">

I started it and clicked on the Initialize button and checked the Podman window dev tools. There are messages repeating in a loop, some containing names of existing containers (not created or maintained by Podman) that I have in my Docker Desktop instance.

This is what I am seeing in the dev tools console:

```
index.ts:1327 main ↪️ Kubeconfig path /Users/tom/.kube/config provided does not exist. Skipping.
index.ts:1327 main ↪️ PluginSystem: received dom-ready event from the UI
index.ts:1327 main ↪️ Delayed startup, flushing 6 events
index.ts:1327 main ↪️ System ready. Loading extensions...
index.ts:1327 main ↪️ Activating extension (compose)
index.ts:1327 main ↪️ Activation extension (compose) ended
index.ts:1327 main ↪️ Activating extension (docker)
index.ts:1327 main ↪️ Activation extension (docker) ended
index.ts:1327 main ↪️ Activating extension (kind)
index.ts:1327 main ↪️ Activating extension (kube-context)
index.ts:1327 main ↪️ [kube-context] starting extension kube-context
index.ts:1327 main ↪️ Activation extension (kube-context) ended
index.ts:1327 main ↪️ Activating extension (lima)
index.ts:1327 main ↪️ [lima] Could not find podman socket at /Users/tom/.lima/podman/sock/podman.sock
index.ts:1327 main ↪️ Activation extension (lima) ended
index.ts:1327 main ↪️ Activating extension (podman)
index.ts:1327 main ↪️ Activating extension (registries)
index.ts:1327 main ↪️ Activation extension (registries) ended
index.ts:1327 main ↪️ Activation extension (podman) ended
index.ts:1327 main ↪️ Retrieved fetchable extensions in 304.05624997615814ms
index.ts:1327 main ↪️ [kind] Error: Error: spawn kind-darwin-arm64 ENOENT
    at ChildProcess.<anonymous> (/Applications/Podman Desktop.app/Contents/Resources/app.asar/extensions/kind/builtin/kind.cdix/dist/extension.js:7:11544)
index.ts:1327 main ↪️ Activation extension (kind) ended
index.ts:1327 main ↪️ PluginSystem: initialization done.
index.ts:1327 main ↪️ Autostarting container engine
index.ts:1327 main ↪️ event is "exec_create: wget --no-verbose --tries=1 --spider http://localhost:9999/health" ID "77c04fdf617be45f2cd668809496693666dcf8e5460dccd2c9b4f295e7a727ff"
index.ts:1327 main ↪️ event is "exec_create: wget --no-verbose --tries=1 --spider http://127.0.0.1:5000/status" ID "b773d50d865ce4c76658948bcbd415162aa98ddc42a99b4ffc67f4e252f39908"
index.ts:1327 main ↪️ event is "exec_start: wget --no-verbose --tries=1 --spider http://localhost:9999/health" ID "77c04fdf617be45f2cd668809496693666dcf8e5460dccd2c9b4f295e7a727ff"
index.ts:1327 main ↪️ event is "exec_start: wget --no-verbose --tries=1 --spider http://127.0.0.1:5000/status" ID "b773d50d865ce4c76658948bcbd415162aa98ddc42a99b4ffc67f4e252f39908"
index.ts:1327 main ↪️ event is "exec_create: bash -c printf \\0 > /dev/tcp/localhost/4000" ID "7f6167f854fb6936dbf356dfcf9a5c32057304b02a14d9627c5c944910286ebf"
index.ts:1327 main ↪️ event is "exec_start: bash -c printf \\0 > /dev/tcp/localhost/4000" ID "7f6167f854fb6936dbf356dfcf9a5c32057304b02a14d9627c5c944910286ebf"
index.ts:1327 main ↪️ event is "exec_die" ID "b773d50d865ce4c76658948bcbd415162aa98ddc42a99b4ffc67f4e252f39908"
index.ts:1327 main ↪️ event is "exec_die" ID "7f6167f854fb6936dbf356dfcf9a5c32057304b02a14d9627c5c944910286ebf"
index.ts:1327 main ↪️ event is "exec_die" ID "77c04fdf617be45f2cd668809496693666dcf8e5460dccd2c9b4f295e7a727ff"
index.ts:1327 main ↪️ event is "exec_create: pg_isready -U postgres -h localhost -p 5432" ID "4b917c9dd33879bb7de777c688c159139026eb049781c7448ca9415537327085"
index.ts:1327 main ↪️ event is "exec_start: pg_isready -U postgres -h localhost -p 5432" ID "4b917c9dd33879bb7de777c688c159139026eb049781c7448ca9415537327085"
index.ts:1327 main ↪️ event is "exec_create: imgproxy health" ID "5ec76798de084eb1b5ab27510a209d3098692fcf2936c23e6573a12f40b021ce"
index.ts:1327 main ↪️ event is "exec_start: imgproxy health" ID "5ec76798de084eb1b5ab27510a209d3098692fcf2936c23e6573a12f40b021ce"
index.ts:1327 main ↪️ event is "exec_die" ID "4b917c9dd33879bb7de777c688c159139026eb049781c7448ca9415537327085"
index.ts:1327 main ↪️ event is "exec_die" ID "5ec76798de084eb1b5ab27510a209d3098692fcf2936c23e6573a12f40b021ce"
```

etc.

It keeps looping like this with `exec_create`, `exec_start` and `exec_die` with matching IDs.
On the high level this then happens for other actions:

```
"exec_create: node -e require('http').get('http://localhost:8080/health', (r) => {if (r.statusCode !== 200) throw new Error(r.statusCode)})" 
"exec_start: node -e require('http').get('http://localhost:8080/health', (r) => {if (r.statusCode !== 200) throw new Error(r.statusCode)})"
"exec_create: node -e require('http').get('http://localhost:3000/api/profile', (r) => {if (r.statusCode !== 200) throw new Error(r.statusCode)})"
"exec_start: node -e require('http').get('http://localhost:3000/api/profile', (r) => {if (r.statusCode !== 200) throw new Error(r.statusCode)})"
"exec_die"
"exec_die"
```

and so on.

There is no one service/component that stands out, everything is just create-start-die forever and eventually the set starts looping.

I have the logs locally so feel free to contact me out of band for the whole logs, I didn't want to copy them to the issue because there are private container names in there.

BTW I have tried to set up Podman twice on two different machines and it failed each time. @benoitf now that there are logs and more info, can you try reproducing on a machine yourself and see if you run into the same issue? This seems quite reproducible at least in my experience.

**Comment by Lewiscowles1986** [comment]
`mkdir -p ~/.lima/podman/sock` was all I had to do to stop the loop when podman had been exited. Hope this helps someone.

**Comment by benoitf** [comment]
Thanks for your report @Lewiscowles1986

**Comment by darkBuddha** [comment]
just had same issue with Podman 1.1.0 on macOS 13.4 (installed via homebrew)

**Comment by evanshortiss** [comment]
EDIT: Check my new comment below this!

I'm running into this issue. I recently upgraded to macOS Ventura on an M1 MacBook Pro. Could the update be related? My Podman installation and machine are working as expected. It's only the Podman Deskop UI that's "stuck".

```
Starting machine "podman-machine-default"
Waiting for VM ...
Mounting volume... /Users/eshortis/:/Users/eshortis/

This machine is currently configured in rootless mode. If your containers
require root permissions (e.g. ports < 1024), or if you run into compatibility
issues with non-podman clients, you can switch using the following command:

	podman machine set --rootful

API forwarding listening on: /var/run/docker.sock
Docker API clients default to this address. You do not need to set DOCKER_HOST.
```

There's no useful information printed in the DevTools as far as I can see:

<img width="1162" alt="Screenshot 2023-06-14 at 3 16 37 PM" src="https://github.com/containers/podman-desktop/assets/1303687/c2606018-3afb-4c89-82a4-8e6682123394">

**Comment by evanshortiss** [comment]
So, it turns out that Podman Desktop finally passed the "Initializing" phase while I was writing my previous comment, but it takes over a minute. 

I noticed that Podman Desktop can start up quickly (a few seconds) when I disable my Pi-Hole DNS. Is Podman Desktop reaching out to an analytics service on startup? If that analytics lookup fails does it stall Podman Desktop? Those are the only blocked DNS lookups in my Pi-Hole logs that correspond to when I launched Podman Desktop. 

EDIT: [This might be the culprit?](https://github.com/containers/podman-desktop/blob/main/packages/main/src/plugin/index.ts#L349) It might also require a separate issue? 😄

**Comment by deboer-tim** [comment]
I put an intentional sleep in the code where the telemetry initializes and indeed it hung on Initializing screen as above. We can fix that and make it async, but I don't think everyone above is hitting this same issue.

**Comment by evanshortiss** [comment]
@deboer-tim I tried to change it to the code below, and it didn't work. The `telemetry.init()` was a red herring, since I found the real fix and opened a [PR explaining it](https://github.com/containers/podman-desktop/pull/2884). TLDR; Podman Dekstop attempts to reach out to the current cluster targeted by your kubeconfig. If this cluster is not available it delays the initialisation!

```ts
// Do not await init() complete, simply move along
telemetry.init().then(() => { console.log('telemetry init success') }).catch((err) => { console.error('telemetry init error', err) });
```

The issue for me [was this line of code](https://github.com/containers/podman-desktop/blob/main/packages/main/src/plugin/kubernetes-client.ts#L171). I noticed in the DevTools logs that I have a log line that states `Error while fetching API groups: Error: connect ETIMEDOUT x.x.x.x:6443`. This appears immediately before the initialisation completes. It seems that Podman Desktop is trying to reaching out to an OpenShift cluster on my mesh network, but it fails because I have the OpenShift instance powered down. Seems to me that this code needs to use more aggressive timeout values? The Kubernetes client used must have an easy way to configure HTTP timeouts, but I wasn't able to see it.

**Comment by deboer-tim** [comment]
Thanks for narrowing this down, Evan. Agreed we need to reduce the timeout, make it async, or likely best: both.

IMHO a delay in sending out the first telemetry event shouldn't block initialization either, but I'll consider that unrelated and follow up independently.

**Comment by evanshortiss** [comment]
Thanks, and agreed! 😃

**Comment by memark** [comment]
Same here. Podman CLI works fine, including running containers. Podman Desktop just hangs with "Initializing".

**Comment by Gauravtalreja1** [comment]
Hello, I've installed the latest podman and podman-desktop 1.3.1 from brew on MacBook M1. I'm also seeing this issue on podman-desktop, able to manage containers using podman from terminal but podman-desktop hangs, and guessing the fix mentioned for async telemetry setup might be included now, so just wondering if this is still being worked on or if there is any workaround?  Thank you in advance :)

**Comment by benoitf** [comment]
@Gauravtalreja1 when you say it hangs, do you see some UI or errors ?

could you open the devtools windows (<kbd>Command</kbd>+<kbd>Option</kbd>+<kbd>i</kbd> )

**Comment by yasirmturk** [comment]
I am not able to connect to the socket 
![Screenshot 2023-09-12 at 16 37 45](https://github.com/containers/podman-desktop/assets/306691/d90d4eb1-4cac-4496-abf1-c3ba1ca11501)

**Comment by lpar** [comment]
Console errors I get:

```
index.ts:1506 main ↪️ Kubeconfig path /Users/meta/.kube/config provided does not exist. Skipping.
index.ts:1506 main ↪️ [lima] Could not find socket at /Users/meta/.lima/podman/sock/podman.sock
index.ts:1506 main ↪️ [lima] Could not find config at /Users/meta/.lima/podman/copied-from-guest/kubeconfig.yaml
main ↪️ [kind] Error: Error: spawn kind-darwin-arm64 ENOENT
    at ChildProcess.<anonymous> (/Applications/Podman Desktop.app/Contents/Resources/app.asar/extensions/kind/builtin/kind.cdix/dist/extension.js:7:11545)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1091:16)
    at Socket.<anonymous> (node:internal/child_process:449:11)
    at Socket.emit (node:events:513:28)
    at Pipe.<anonymous> (node:net:322:12)
```

**Comment by abdullahcicekli** [comment]
Any progress ?

**Comment by sofpodev** [comment]
It also happens on Mac Intel (14.1). It also causes Mac to freeze!
Any permanent solution or workaround?

**Comment by Jap8nted** [comment]
Same here, 

I am using Mac M1 arm 

Tried using Podman command line and desktop, was not able to use it. Still don't understand what is the problem, there are multiple GitHub issues talking about this but none of them really gives a solution for this. 


<img width="1483" alt="image" src="https://github.com/containers/podman-desktop/assets/5432705/35734131-3fdf-48ae-a2ed-b9fbdb6b93cb">

**Comment by benoitf** [comment]
@Jap8nted could you go to settings (cog in left navbar) then resources then click on the podman machine and click on ↗️ icon to get podman machine details

and here go to the logs tab

**Comment by Jap8nted** [comment]
Thanks for your response @benoitf , I was installing using:

`brew install podman-desktop`

I tried once installing the downloaded file from https://podman-desktop.io/downloads, it worked fine. 

I have to fix some issues, after I do that, will try to reproduce it and upload the logs!

**Comment by Janhouse** [comment]
I did have an older vm running fine, but the podman-machine-default_fedora-coreos-39.20240112.2.0-qemu.aarch64.qcow2 image crashes when starting it.
```sh
podman machine init
podman machine start --log-level=trace
```
<img width="651" alt="image" src="https://github.com/containers/podman-desktop/assets/1036439/498ab5e0-ef51-4d88-8f9c-4ed801fdb287">

**Comment by benoitf** [comment]
the recent comment was about the Qemu 8.2.0 issue and ARM processor/Silicon (fixed with qemu 8.2.1)
https://gitlab.com/qemu-project/qemu/-/issues/1990

else it looks we do not reproduce it, closing

**Comment by folex** [comment]
Just installed from brew. 1.7.1 was working fine, updated to 1.12, and then reinstalled 1.12 from brew.

<img width="1840" alt="image" src="https://github.com/user-attachments/assets/6a581e40-851c-44c4-8b68-f915350d6c8f">
<img width="396" alt="image" src="https://github.com/user-attachments/assets/44390b4c-3143-405a-8a58-1f1d2a2ed553">


After ~10 mins, restarted to see suggestion to initalize and start again. 

<img width="1162" alt="image" src="https://github.com/user-attachments/assets/219f7970-0304-424e-81ed-27468bcf8870">

If I go to Set Up, it hangs on creating maching 
<img width="1840" alt="image" src="https://github.com/user-attachments/assets/998fb764-0eec-47a6-84ce-4640ebf880ab">


`podman machine init` also hangs.
